test/e2e/egress_test.go

```diff
@@ -38,7 +38,15 @@ import (
 	controllerconfig "antrea.io/antrea/pkg/config/controller"
 )
 
-const waitEgressRealizedTimeout = 3 * time.Second
+const (
+	egressRealizedTimeout = 5 * time.Second
+	// When controller and agent restarted, agent may fail to start watch for EgressGroup,
+	// because antrea service is still not reachable for the agent.
+	// Thus agent wait another 5 second and try to start watch for EgressGroup again.
```

Comment:
If it's caused by antrea-agent and antrea-controller restarts, it should just wait for antrea-agent and antrea-controller to be ready before starting any test, then we could still expect the tests to finish in a reasonable timeout.

`mutateAntreaConfigMap` has waited for components to be ready if they are restarted, but controller is restarted after agent, so the readiness probe result of agent may be out-of-date. Maybe we could swap the order of restarting controller and agent after #3413 is merged. And the state of connection to egressGroup API should be part of `ReadyzChecks` when egress feature is on.